internal/adapters/operation_lease/redis/redis.go

```diff
@@ -64,3 +86,15 @@ func (r *redisOperationLeaseStorage) FetchLeaseTTL(ctx context.Context, schedule
 func (r *redisOperationLeaseStorage) ListExpiredLeases(ctx context.Context, schedulerName string, maxLease time.Time) ([]operation.OperationLease, error) {
 	return nil, nil
 }
+
+func (r *redisOperationLeaseStorage) buildSchedulerOperationLeaseKey(schedulerName string) string {
+	return fmt.Sprintf("operations:%s:operationsLease", schedulerName)
+}
+
+func (r *redisOperationLeaseStorage) existsOperationLease(ctx context.Context, schedulerName, operationId string) (bool, error) {
+	operationLeaseList, _, err := r.client.ZScan(ctx, r.buildSchedulerOperationLeaseKey(schedulerName), 0, operationId, 0).Result()
```

Comment:
@caiordjesus WDYT about using ZSCORE for checking the existence here? it's time complexity is constant in comparison with Zscan that have a linear complexity in worst case. Ref: https://redis.io/commands/zscore

Reply:
```
operationLeaseScore, err := client.ZScore(ctx, r.buildSchedulerOperationLeaseKey(schedulerName), operationID).Result()
```

Reply:
se o erro for diferente de nil, significa que o valor que estamos buscando não existe ainda, então você poderia retornar apenas err != nil

Reply:
Awesome insight!!
Just made the change!